Earthfile

```diff
@@ -191,6 +195,26 @@ sbom-generate:
     # Save the result back to the host:
     SAVE ARTIFACT /s/cyclonedx.sbom.json AS LOCAL etc/cyclonedx.sbom.json
 
+# sbom-download :
+#   Download an augmented SBOM from the Silk server for the given branch. Exports
+#   the artifact as /augmented-sbom.json
+#
+# Requires credentials for silk access.
+sbom-download:
+    FROM alpine:3.20
+    ARG --required branch
+    # Run the SilkBomb tool to download the artifact that matches the requested branch
+    FROM +silkbomb
+    # Set --no-cache, because the remote artifact could change arbitrarily over time
+    RUN --no-cache \
+        --secret SILK_CLIENT_ID \
+        --secret SILK_CLIENT_SECRET \
+        silkbomb download \
+            --sbom-out augmented-sbom.json \
+            --silk-asset-group mongo-c-driver-${branch}
+    # Export as /augmented-sbom.json
+    SAVE ARTIFACT augmented-sbom.json
```

Comment:
Should this be saved to `etc/augmented-sbom.json` so it is placed alongside the SBOM Lite document?

Aside: I'm considering naming the Augmented SBOM `etc/augmented.sbom.json` for the CXX Driver. Thoughts? (There doesn't seem to be a standard/recommended naming scheme for this file.)

Reply:
IIUC, the augmented SBOM should be generated on-the-fly (since it can be updated asynchronously) as part of a release artifact, so it shouldn't matter where it goes. If my understanding is incorrect, please correct me.

I have no strong opinion on the naming of the file itself. In this case I just named it by whatever felt intuitive at the moment. If it _is_ an ephemeral artifact, then the name can be changed arbitrarily in the future.

Overall I don't like the way this particular artifact is generated and handled, but that may be unavoidable.

Reply:
> should be generated on-the-fly [...] so it shouldn't matter where it goes

Correct, although if/when we eventually do distribute it as part of a release, it may be preferable to place it in a consistent location, thus my suggestion to place it under `etc` rather than project root.

Reply:
Ah. There's some confusion. The `SAVE ARTIFACT` command does not place any files in the host, it only declares a file/directory to be exported so that they can be copied onto the host (with `earthly --artifact`) or into another target. The `+sbom-generate` target _does_ write a file into the host with the `AS LOCAL` clause. (Refer: https://docs.earthly.dev/docs/earthfile#save-artifact).

Reply:
Ah, I see. Then there should be nothing to do for this PR. My suggestion should be handled by followup work. SGTM. 👍